sound/soc/sof/intel/hda-dai-ops.c

```diff
@@ -258,6 +370,11 @@ void hda_select_dai_widget_ops(struct snd_sof_dev *sdev, struct snd_sof_widget *
 			struct hda_dai_widget_ops *ops = &hda_dai_ops;
 
 			sdai->private_ops = ops;
+			dma_ops = ops->dma_ops;
+
+			dma_ops->pre_trigger = hda_ipc4_pre_trigger;
+			dma_ops->trigger = hda_trigger;
+			dma_ops->post_trigger = hda_ipc4_post_trigger;
```

Comment:
@ranj063, this is definitely not going to work as intended.
You can not have separate callbacks defined for different DAIs/uses as you are modifying a single `hda_dai_ops` and it's single `hda_dma_ops` if a single DAI would need different ops, you will change it for all DAIs.

You need all this in static or do a kmemdump and modify the copy.

Reply:
not following @ujfalusi's comment